src/main/java/net/apnic/whowas/intervaltree/Interval.java

```diff
@@ -7,7 +7,7 @@
     default int compareTo(Interval<K> other) {
         int lc = low().compareTo(other.low());
         if (lc == 0) {
-            lc = high().compareTo(other.high());
+            lc = -1 * high().compareTo(other.high());
```

Comment:
{suggestion} Adding a comment about the interval comparison here may save my head again later :)

Reply:
Agreed

Reply:
Done & Done